pkgs/os-specific/linux/bcc/default.nix

```diff
@@ -59,6 +62,10 @@ python.pkgs.buildPythonApplication rec {
       --replace '$'{exec_prefix}/@CMAKE_INSTALL_LIBDIR@ @CMAKE_INSTALL_FULL_LIBDIR@
   '';
 
+  preInstall = ''
+    # required for setuptool during install
+    export PYTHONPATH=$out/${python3.sitePackages}:$PYTHONPATH
+  '';
```

Comment:
This could probably be fixed by constructing a python env

Reply:
> /home/x/nixpkgs/outputs/out/lib/python3.10/site-packages/

What is that path? Are you not using the sandbox?

Reply:
I was building with `nix -L build -f . bcc` from nixpkgs' directory; `/home/x/nixpkgs` was the nixpkgs directory (I removed the user bit), on a nixos unstable linux machine.

I'm almost certain I didn't disable the sandbox (I actually have `nix.settings.sandbox = true;` from before it was default but I think it's not required anymore now...?) , but I guess it's the path that was derived from whatever cmake install target the build script derived.
But yeah I'm surprised it contained my home path and not /build or whatever builder sandbox path should be using...

I retried building just now, printing `$out` and it appears correct, and so is the python path printed when I built just now which has the same warnings with paths I'd expect:
```
bcc> $out is /nix/store/ijf86wzk41fdkly3slfvkr71dv6lbfk6-bcc-0.28.0
[...]
bcc> Checking .pth file support in /nix/store/ijf86wzk41fdkly3slfvkr71dv6lbfk6-bcc-0.28.0/lib/python3.10/site-packages/
bcc> /nix/store/1r6n7v2wam7gkr18gxccpg7p5ywgw551-python3-3.10.12/bin/python3 -E -c pass
bcc> TEST FAILED: /nix/store/ijf86wzk41fdkly3slfvkr71dv6lbfk6-bcc-0.28.0/lib/python3.10/site-packages/ does NOT support .pth files
bcc> bad install directory or PYTHONPATH
bcc> You are attempting to install a package to a directory that is not
bcc> on PYTHONPATH and which Python does not read ".pth" files from.  The
bcc> installation directory you specified (via --install-dir, --prefix, or
bcc> the distutils default setting) was:
bcc>     /nix/store/ijf86wzk41fdkly3slfvkr71dv6lbfk6-bcc-0.28.0/lib/python3.10/site-packages/
bcc> and your PYTHONPATH environment variable currently contains:
bcc>     '/nix/store/1r6n7v2wam7gkr18gxccpg7p5ywgw551-python3-3.10.12/lib/python3.10/site-packages:/nix/store/cr689gv3clfqhbl05pfqd9g9khzdidcb-python3.10-setuptools-67.4.0/lib/python3.10/site-packages:/nix/store/mss4z8ivvpl5px7jvwhb8gfxs7m1c0ym-python3.10-netaddr-0.8.0/lib/python3.10/site-packages'
bcc> Here are some of your options for correcting the problem:
bcc> * You can choose a different installation directory, i.e., one that is
bcc>   on PYTHONPATH or supports .pth files
bcc> * You can add the installation directory to the PYTHONPATH environment
bcc>   variable.  (It must then also be on PYTHONPATH whenever you run
bcc>   Python and want to use the package(s) you are installing.)
bcc> * You can set up the installation directory to support ".pth" files by
bcc>   using one of the approaches described here:
bcc>   https://setuptools.pypa.io/en/latest/deprecated/easy_install.html#custom-installation-locations
```

So, err, not quite sure what happened, and any logs that might have remained on filesystem have long been gc'd so /shrug
(I also tried various commits from reflog to check and not noticing a difference, but that won't be downgrading my 'nix' version to the os I was running at the time)


Either way the error remains the same: without setting the variable, the package builds but the python install step failed and the resulting egg file is unusable from python import directives

Reply:
I would be fine with this solution. python env might be actually more code and currently I don't see the advantage of it.